Update 'rv' value generation based on randomness flag + editorial changes to improve clarity (#261)
open-telemetry/oteps: text/trace/0235-sampling-threshold-in-trace-state.md

2. The rejection threshold: the 56-bit `T` value.

If `R` >= `T`, *keep* the span, else *drop* the span.

`T` represents the maximum threshold that was applied in all previous consistent sampling stages. If the current sampling stage applies a greater threshold value than any stage before, it MUST update (increase) the threshold correspondingly.

## Explanation

### Example

Here is an example involving three participating operations `A`, `B`, and `C`:

`A` -> `B` -> `C`

where -> indicates a parent to child relationship.

- `A` uses consistent probability sampling with a sampling probability of 0.25. This corresponds to a rejection threshold of (1-0.25)*2^56 = 0xc0000000000000.
- `B` uses consistent probability sampling with a sampling probability of 0.5. This corresponds to a rejection threshold of (1-0.5)*2^56 = 0x80000000000000.
- `C` uses a parent-based sampler. This means that C indirectly follows the same rejection threshold as its parent B.

When the sampling decision for `A` is to *keep* the span, its outgoing traceparent will have the `sampled` flag SET and the `th` in its outgoing tracestate will be set to `0xc0_0000_0000_0000`.

> When the sampling decision for `A` is to *drop* the span, its outgoing traceparent will have the `sampled` flag UNSET but the `th` in its outgoing tracestate will still be set to `0xc0_0000_0000_0000`.

Reviewer: ```suggestion
When the sampling decision for `A` is to *drop* the span, its outgoing traceparent will have the `sampled` flag UNSET and `th` in its outgoing tracestate will be removed.
```

Shouldn't the `th` value be removed in this case? There is no need to propagate the `th`-value if the sampling decision is negative. Otherwise, we would have to define a way to encode a threshold of 2^56 corresponding to a 0% probability.

Sampling Decisions MUST be propagated by setting the value of the `th` key in the Tracestate header according to the above.

> This design requires that as a given span progresses along its collection path, `th` is non decreasing (and, in particular, must be increased at stages that apply lower sampling probabilities).

Reviewer 2: This paragraph and the next one seem to imply that increasing T/th is fine, but decreasing it is not, whereas the Head Samplers sections puts this as an actual example.

Reviewer 2: Ping @kalyanaj
